src/aws_s3_presigned_url.erl

```diff
@@ -23,23 +23,22 @@ make_presigned_v4_url(Client0, Method, ExpireSeconds, Bucket, Key) ->
     Client = Client0#{service => <<"s3">>},
     SecurityToken = maps:get(token, Client, undefined),
     AccessKeyID = maps:get(access_key_id, Client),
+    SecretAccessKey = maps:get(secret_access_key, Client),
     Region = maps:get(region, Client),
+    Service = maps:get(service, Client),
     Host = build_host(<<"s3">>, Client, Bucket),
-    {BaseHost, URL0} = build_url(Host, Path, Client, Bucket),
+    URL = build_url(Host, Path, Client, Bucket),
     Now = calendar:universal_time(),
-    ShortDate = list_to_binary(ec_date:format("Ymd", Now)),
-    Credential = make_credential(AccessKeyID, ShortDate, Region),
-    Headers0 = [{<<"Host">>, Host},
-                {<<"X-Amz-Algorithm">>, <<"AWS4-HMAC-SHA256">>},
-                {<<"X-Amz-Credential">>, Credential},
-                {<<"X-Amz-Date">>, list_to_binary(ec_date:format("YmdTHisZ", Now))},
-                {<<"X-Amz-Expires">>, integer_to_binary(ExpireSeconds)},
-                {<<"X-Amz-SignedHeaders">>, <<"host">>}
+    Options0 = [ {ttl, ExpireSeconds}
+               , {body_digest, <<"UNSIGNED-PAYLOAD">>}
```

Comment:
Since the old implementation was sending in `UNSIGNED-PAYLOAD` as the payload (See the !Important box at [sig-v4-header-based-auth](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html) I've kept this. I'm not sure if clients may already use presigned urls but figured I'd keep the same signature of the URL in place.